rhythm.py

```diff
@@ -1,24 +1,81 @@
+import logging
+from abc import ABCMeta, abstractmethod
+from time import sleep
+
 from regression import calculate_regression
-import math
 
-HANDSTROKE_GAP = 1
 
-class Rhythm:
-    logger_name = "RHYTHM"
+class Rhythm(metaclass=ABCMeta):
+    @abstractmethod
+    def wait_for_bell_time(self, current_time, bell, row_number, place, user_controlled, stroke):
```

Comment:
I ended up with more parameters here that I intended, but they all currently seem needed...